### ISSUE TITLE
Unable to set Image - JPackage

### ISSUE THREAD

**Issue by Srijan-Sengupta** [beryx-gist/badass-jlink-example#2]
I am unable set the icon image of my application for jpackage. How must i set it?

**Comment by siordache** [comment]
If you use org.beryx.jlink 2.22.0, see [this example](https://github.com/beryx-gist/badass-jlink-example-richtextfx/blob/7cab227f75945d7d617ff7569c1bd242be5a8afa/build.gradle#L36-L38).

**Comment by Srijan-Sengupta** [comment]
Thank you very much but I a have another Question : 
**How may I configure the .desktop file and add an attribute: StartupWMClass=com.srijan ?**

**Comment by Srijan-Sengupta** [comment]
Secondly the deb file application shortcut does not have my selected icon what must I do?

**Comment by siordache** [comment]
I'm not familiar with .desktop files but from what I see [here](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#recognized-keys), I assume it should be something like this:

```
[Desktop Entry]
Type=Application
StartupWMClass=com.srijan
Name=...
Icon=...
...
```

I'm on holiday now and not able to check if it works, but you probably need to put the .desktop file in the src/main/resources directory and configure the resource-dir:
```
jpackage {
    ...
    installerOptions += ['--resource-dir', "build/jpackage/your-app-name/lib"]
    ...
}
```


---

To investigate why the shortcut does not have your icon, add this:
 
```
jpackage {
    ...
    installerOptions += ['--verbose']
    ...
}
```
and execute:
```
./gradlew -i clean jpackage
```

Then look for messages about using a default package resource (as [described here](https://docs.oracle.com/en/java/javase/14/jpackage/override-jpackage-resources.html#GUID-62412365-821D-4736-A20D-70F2D47136D8)).

**Comment by Srijan-Sengupta** [comment]
Thank you very much. Using verbose I have been able to set the icon of my application but sadly I am still unable to configure the .desktop file.

**Comment by Srijan-Sengupta** [comment]
Today while experimenting, accidentally I found my solution to this.

I just set my `--resource-dir` to `$project.root/src/main/resources` and there I placed my icon, desktop cofiguration file, etc. and JPackage automatically picked up all the resources needed.

As, I have resolved this issue with help of your references, I am closing this issue.
Thank you very much, your resources helped me a lot. But I would request you to configure the build.gradle because according to this [documentation](https://docs.oracle.com/en/java/javase/14/jpackage/override-jpackage-resources.html#GUID-1B718F8B-B68D-4D46-B1DB-003D7729AAB6), I feel it is better to set the resources dir to our resources directory.

Thanks a lot and a happy holiday.
